### PR TITLE
rhbz1048293 Fix liquibase changeset to be compatible with mysql/mariadb version > 5.5

### PR DESCRIPTION
**Pull request by seanf**
https://bugzilla.redhat.com/show_bug.cgi?id=1048293

### CODE REVIEW

File: zanata-war/src/main/resources/db/changelogs/db.changelog-1.5.xml [hunk 1]
```diff
@@ -134,13 +134,24 @@
...
+        <dropPrimaryKey tableName="HTermComment"/>
         <addAutoIncrement tableName="HGlossaryEntry" columnName="id" columnDataType="bigint"/>
         <addAutoIncrement tableName="HGlossaryTerm" columnName="id" columnDataType="bigint"/>
         <addAutoIncrement tableName="HTermComment" columnName="id" columnDataType="bigint"/>
+        <addPrimaryKey tableName="HGlossaryEntry" columnNames="id" />
+        <addPrimaryKey tableName="HGlossaryTerm" columnNames="id" />
+        <addPrimaryKey tableName="HTermComment" columnNames="id" />
+        <addForeignKeyConstraint baseTableName="HGlossaryTerm"
+                baseColumnNames="glossaryEntryId" constraintName="UKglossaryterm_glossary_entry_id"
+                referencedTableName="HGlossaryEntry" referencedColumnNames="id" />
```
Comment (davidmason) [inline zanata-war/src/main/resources/db/changelogs/db.changelog-1.5.xml]: Can you add an xml comment with the reason for removing then adding the keys and constraints? I assume that the `addAutoIncrement` lines cannot be applied while the keys and constraints are in place, but we should not have to make such an assumption.

Comment (seanf) [inline zanata-war/src/main/resources/db/changelogs/db.changelog-1.5.xml]: Good point.  Done.

File: functional-test/pom.xml [hunk 2]
```diff
@@ -36,6 +36,9 @@
     <ds.connection.url>jdbc:mysql://localhost:${mysql.port}/${ds.database}?characterEncoding=UTF-8</ds.connection.url>
     <ds.driver.class>com.mysql.jdbc.Driver</ds.driver.class>
 
+    <mysql.dist.version>5.5.34</mysql.dist.version>
+    <!-- <mysql.dist.version>5.6.21</mysql.dist.version> -->
```
Comment (davidmason) [inline functional-test/pom.xml]: Did you purposely just modify this?

Comment (davidmason) [inline functional-test/pom.xml]: :+1:

Comment (seanf) [inline functional-test/pom.xml]: I'm trying to test against both versions, and for the moment it's easier than switching the build params in jenkins.

Comment (davidmason) [inline functional-test/pom.xml]: Fair enough. Ping me if anything else changes and it needs another review.

**Comment by seanf** [comment]
@davidmason I have modified the changeset again.  It now works on 5.6.14, just trying 5.5 again now.

**Comment by seanf** [comment]
@davidmason sorry, modified the changeset again.  I just had to add another `validCheckSum` to make it work on existing databases.  Looks like this is a reasonable procedure: https://wiki.openmrs.org/display/docs/How+to+Modify+a+Bad+Changeset

**Comment by davidmason** [comment]
:+1:

**Comment by seanf** [comment]
@djansen-redhat Are you taking this one over?

**Comment by djansen-redhat** [comment]
:white_check_mark: Tested. (Fedora 21, mariadb 10.0.17-1, Wildfly 8.2)
Ran into an issue with installing mariadb as a service, as well as an issue with duplicate mysql-connector-java packages which was resolved by removing the one in JBoss deployment, yum installing it and symlinking it to the deployments dir.

**Comment by seanf** [comment]
> Ran into an issue with installing mariadb as a service, as well as an issue with duplicate mysql-connector-java packages which was resolved by removing the one in JBoss deployment, yum installing it and symlinking it to the deployments dir.


@djansen-redhat that sounds like a problem with zanata-3.6.0-wildfly-8.1.zip, is that right?

Are you saying you had duplicate mysql-connector-java packages, even before you put in the symlink?

**Pull request merged**